internal/exec/util/blkid.c

```diff
@@ -141,10 +141,26 @@ result_t blkid_get_partition(const char *device, int part_num, struct partition_
 	if (!part)
 		return RESULT_BAD_INDEX;
 
-	return extract_part_info(part, info);
+	blkid_topology topo = blkid_probe_get_topology(pr);
```

Comment:
I got lost while trying to double-check if this pointer requires to be freed, but I think it doesn't as its lifetime belongs to `pr`. If that is true, a comment for casual readers may be useful.

Reply:
> lifetime
Your Rust is showing. But yeah, it is, I'll add a comment.